core/src/main/java/org/elasticsearch/common/util/CancellableThreads.java

```diff
@@ -103,7 +104,7 @@ public void execute(Interruptable interruptable) {
                 onCancel(reason, throwable);
             } else if (throwable != null) {
                 // if we're not canceling, we throw the original exception
-                throw throwable;
+                ExceptionsHelper.reThrowIfNotNull(throwable);
```

Comment:
This can just be `throw throwable` since we check if it's not null immediately above? Or else we can remove the check and make it just an `else`